src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final double CAPACITY_FACTOR = 0.75;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] entryset;
+    private int size;
+
+    public MyHashMap() {
+        this.entryset = new Node[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (((double) size / entryset.length) >= CAPACITY_FACTOR) {
+            resize();
+        }
+
+        int bucket = getBucket(key);
+        if (entryset[bucket] == null) {
+            entryset[bucket] = new Node<>(key, value);
+            size++;
+        } else {
+            Node<K, V> entry = entryset[bucket];
+
+            do {
+                if (Objects.equals(entry.key, key)) {
+                    entry.value = value;
+                    return;
+                }
+                if (entry.next == null) {
+                    break;
+                }
+            } while ((entry = entry.next) != null);
+
+            entry.next = new Node<>(key, value);
+            size++;
```

Comment:
```suggestion
            Node<K, V> entry = entryset[bucket];
            
            while (entry != null) {
                if (Objects.equals(entry.key, key)) {
                    entry.value = value;
                    return;
                }
                entry = entry.next;
            }
            entrySet[bucket] = new Node<>(key, value, bucket[index]);
            size++;
```
in general do while is not a good practice
while is easier to read